### PR TITLE
#53613: Eltwise Unary bfloat16 coverage tests - category 1 ops

### PR DESCRIPTION
**Pull request by KalaivaniMCW**
### PR Category

Test Only

### Summary

#53613 

### Notes for reviewers

Eltwise unit tests restructuring
- Add `test_unary_category1_bfloat16.py` with exhaustive bfloat16 bitpattern coverage for 47 category-1 unary ops (trig, hyperbolic, comparison, rounding, special math, logical, and utility)
- Add shared test utilities in `eltwise_test_utils.py` (`generate_bfloat16_bits`, `generate_bfloat16_bits_in_range`, `flush_to_zero`)
- Each op is tested against its golden function over every representable bfloat16 value in its valid domain, with appropriate tolerance (exact match, ULP=1, or PCC≥0.999 depending on kernel accuracy)
- Remove redundant tests from existing files that are now subsumed by the exhaustive coverage

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:kalaivani/unary_cat1_bf16)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:kalaivani/unary_cat1_bf16)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:kalaivani/unary_cat1_bf16)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:kalaivani/unary_cat1_bf16)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:kalaivani/unary_cat1_bf16)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:kalaivani/unary_cat1_bf16)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:kalaivani/unary_cat1_bf16)

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/eltwise/eltwise_test_utils.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# SPDX-FileCopyrightText: © 2025 Tenstorrent USA, Inc.
```
Comment (Aswinmcw) [inline tests/ttnn/unit_tests/operations/eltwise/eltwise_test_utils.py]: New file added in 2026 — SPDX year should be 2026. The other new file in this PR (`test_unary_category1_bfloat16.py`) already uses 2026.

File: tests/ttnn/unit_tests/operations/eltwise/test_unary_category1_bfloat16.py [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+def test_lgamma(device):
+    high = 1000
+    low = -1000
+
+    # All 2^16 bfloat16 bit patterns (256x256), flattened for masking
+    input_tensor = generate_bfloat16_bits_in_range(low, high).flatten()
+    input_tensor = flush_subnormal_values_to_zero(input_tensor)
+    input_tensor_f32 = input_tensor.to(torch.float32)
+
+    in_range = (input_tensor_f32 >= low) & (input_tensor_f32 <= high)
```
Comment (Aswinmcw) [inline tests/ttnn/unit_tests/operations/eltwise/test_unary_category1_bfloat16.py]: `generate_bfloat16_bits_in_range(low, high)` has already filtered to `[low, high]`, so `in_range` is always all-True and the mask reduces to `~is_non_positive_int`. The comment on 585 is stale too — the helper returns an `(N, 32)` filtered tensor, not the full 256x256 set. Separately, the section comment says lgamma is "defined for x > 0" but the range here is `[-1000, 1000]`.

File: tests/ttnn/unit_tests/operations/eltwise/test_unary_category1_bfloat16.py [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+# ─────────────────────────────────────────────────────────────────────────────
+# Zero comparison and special value checks
+# ─────────────────────────────────────────────────────────────────────────────
+
+
+@pytest.mark.parametrize(
+    "ttnn_op",
+    (ttnn.eqz, ttnn.nez, ttnn.gtz, ttnn.ltz, ttnn.gez, ttnn.lez),
+)
+@pytest.mark.parametrize("ttnn_dtype", [ttnn.bfloat16, ttnn.float32])
```
Comment (Aswinmcw) [inline tests/ttnn/unit_tests/operations/eltwise/test_unary_category1_bfloat16.py]: `ttnn.float32` in a file scoped to bfloat16 — #53613 asks for dtype segregation, so this parametrization probably belongs in the fp32 category file. These six ops are also already covered exhaustively by `test_comparison_to_zero_ops`; the only genuinely new coverage here is the +/-0 / +/-inf / NaN row, which `include_spl_values=True` would give you there.

File: ttnn/ttnn/operations/unary.py [hunk 4]
```diff
@@ -272,6 +272,15 @@ def _golden_function_reciprocal(input_tensor_a, *args, device, **kwargs):
 ttnn.attach_golden_function(ttnn.reciprocal, golden_function=_golden_function_reciprocal)
 
 
+def _golden_function_i1(input_tensor_a, *args, **kwargs):
+    import torch
+
+    return torch.special.i1(input_tensor_a)
+
+
+ttnn.attach_golden_function(ttnn.i1, golden_function=_golden_function_i1)
```
Comment (Aswinmcw) [inline ttnn/ttnn/operations/unary.py:281]: `ttnn.i1` clamps inputs to +/-88.5 on device (see `test_i1_clamp_boundary` in `test_unary_i1.py`), but this golden doesn't. Anything resolving `get_golden_function(ttnn.i1)` with inputs beyond that range — sweeps included — will now report a false mismatch. Worth clamping in the golden to match the kernel.

**Review by Aswinmcw — APPROVED** [review]
LGTM — a few nits below, none blocking.

File: tests/ttnn/unit_tests/operations/eltwise/test_unary_category1_bfloat16.py (hunk 3 above)
Comment (KalaivaniMCW) [reply tests/ttnn/unit_tests/operations/eltwise/test_unary_category1_bfloat16.py]: yes. For the sake of reducing code duplicity it was added to a single file

File: ttnn/ttnn/operations/unary.py (hunk 4 above)
Comment (KalaivaniMCW) [reply ttnn/ttnn/operations/unary.py:281]: The valid domain needs to be documented on our end but I wouldn't suggest modifying the golden, as the torch fn has a broader coverage.

**Review by VirdhatchaniKN — APPROVED** [review]
LGTM
